python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -60,14 +60,17 @@ def __init__(
             ObjectRef[ObjectRefGenerator], Tuple[_TaskState, ray.actor.ActorHandle]
         ] = {}
         # A pool of running actors on which we can execute mapper tasks.
-        self._actor_pool = _ActorPool()
+        self._actor_pool = _ActorPool(autoscaling_policy._config.max_tasks_in_flight)
```

Comment:
Ah good catch! I wish I had gotten to the `ActorPoolMapOperator` unit tests to catch silly bugs like this. Thanks for covering this with the e2e streaming executor test, we can look at adding some `ActorPoolMapOperator` unit tests at some point in the next few weeks.